libmcount/plthook.c

```diff
@@ -246,7 +296,10 @@ static int find_got(struct uftrace_elf_data *elf, struct uftrace_elf_iter *iter,
 
 	memset(&pd->dsymtab, 0, sizeof(pd->dsymtab));
 	/* do not demangle symbol names since it might call dlsym() */
-	load_elf_dynsymtab(&pd->dsymtab, elf, pd->base_addr, 0);
+	if (!plt_found)
```

Comment:
This condition cannot be true here, right?

Reply:
I don't think so. Even modern compilers and x86_64 always retain some PLT entry, we can't exclude the case that the compiler does not retain the PLT entry and totally relies on the GOT table directly. So this `if` expression should be there IMO.

Reply:
> This condition cannot be true here, right?

I don't think so. Even modern compilers and x86_64 always retain some PLT entry, we can't exclude the case that the compiler does not retain the PLT entry and totally relies on the GOT table directly. So this if expression should be there IMO.

Reply:
Hmm.. but I think you checked it on line 265 and returned.

Reply:
Oh now I understand what you meant and it makes sound. I'll revert this diff.